README
NAME

    Dist::Zilla::Plugin::INI::Baked - Add a baked version of your
    configuration to tree automatically

VERSION

    version 0.002002

SYNOPSIS

      # somewhere in dist.ini or even your bundle
      [INI::Baked]
      ; filename        = dist.ini.baked
      ; source_filename = dist.ini
    
      # and and
      dzil build
    
      # and and
      cat $MYDIST/dist.ini.baked  # yay

    Whether you wish to

      * Copy that file back to root/

      * Name that file dist.ini

      * Add/Not add the original dist.ini to your built code.

    All these choices are your discretion, and are presently expected to
    master other dzil plugins to make this possible.

    I recommend:

      * [CopyFilesFromBuild]

      * [CopyFilesFromRelease]

      * Passing exclude rules to [Git::GatherDir]

      * Passing exclude rules to [GatherDir]

    These will of course all still work, because source_filename is read
    directly from $zilla->root

    Patches to make it read from $zilla->files will be accepted, but YAGNI
    for now.

METHODS

 gather_files

    This module subscribes to the -FileGatherer role.

    As such, this module injects a FromCode object during the gather phase.

ATTRIBUTES

 filename

    The name of the file to emit.

    DEFAULT:

      dist.ini.baked

 source_filename

    The name of the file to read

    DEFAULT:

      dist.ini

AUTHOR

    Kent Fredric <kentnl@cpan.org>

COPYRIGHT AND LICENSE

    This software is copyright (c) 2014 by Kent Fredric
    <kentfredric@gmail.com>.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Release history for Dist-Zilla-Plugin-INI-Baked

0.002003 2017-03-07T16:38:54Z 7d9394b

0.002002 2017-03-07T16:38:20Z 8c695d1
 - Bugfix: Avoid test failures with -Ddefault_inc_excludes_dot
 - Remove use of ConfigDumper util
 - Remove use of Test::KENTNL

 [Dependencies::Stats]
 - Dependencies changed since 0.002001, see misc/*.deps* for details
 - configure: (recommends: ↓1)
 - develop: +8 ↑4 -1 (recommends: ↓1, suggests: ↑2)
 - runtime: -1 (recommends: +2)
 - test: ↓1 -1 (recommends: ↓2)

0.002001 2014-12-14T07:19:50Z a292fd7
 [Bugfix]
 - Adjust test to only operate with latest ExpandINI
 - Adjust test to expect comment-copy

 [Dependencies::Stats]
 - Dependencies changed since 0.002000, see misc/*.deps* for details
 - configure: +1 (recommends: ↑1)
 - develop: +5 ↑1 -2 (recommends: +1, suggests: ↑2)
 - runtime: ↓1 -1
 - test: +2 (recommends: +1 ↑2)

0.002000 2014-09-02T08:08:38Z
 [00 Minor]
 - Add dump_config
 - tests refactored.
 - metadata

 [Dependencies::Stats]
 - Dependencies changed since 0.001001, see misc/*.deps* for details
 - develop: +1 ↑1 (suggests: ↑2)
 - runtime: +1
 - test: +1

 [Metadata]
 - Set author = kentnl

 [Tests]
 - implemented now in dztest

0.001001 2014-08-15T21:16:43Z
 [00 Trivial]
 - CPANDAY!
 - no code changes.
 - whitespace reduced.

 [Dependencies::Stats]
 - Dependencies changed since 0.001000, see misc/*.deps* for details
 - develop: +1 (suggests: ↑1)

 [Misc]
 - Improve source side POD to reduce build side whitespace.

0.001000 2014-08-03T06:41:00Z
 - First version.


maint/perlcritic.rc.gen.pl
#!/usr/bin/env perl
## no critic (Modules::RequireVersionVar)

# ABSTRACT: Write an INI file from a bundle

use 5.008;    # utf8
use strict;
use warnings;
use utf8;

our $VERSION = 0.001;

use Carp qw( croak carp );
use Perl::Critic::ProfileCompiler::Util qw( create_bundle );
use Path::Tiny qw(path);

## no critic (ErrorHandling::RequireUseOfExceptions)
my $bundle = create_bundle('Example::Author::KENTNL');
$bundle->configure;

my @stopwords = (
  qw(
    YAGNI dzil
    )
);
for my $wordlist (@stopwords) {
  $bundle->add_or_append_policy_field( 'Documentation::PodSpelling' => ( 'stop_words' => $wordlist ) );
}

#$bundle->add_or_append_policy_field(
#  'Subroutines::ProhibitCallsToUndeclaredSubs' => ( 'exempt_subs' => 'String::Formatter::str_rf' ), );

#$bundle->remove_policy('ErrorHandling::RequireUseOfExceptions');
#$bundle->remove_policy('CodeLayout::RequireUseUTF8');
#$bundle->remove_policy('ErrorHandling::RequireCarping');
#$bundle->remove_policy('NamingConventions::Capitalization');

my $inf = $bundle->actionlist->get_inflated;

my $config = $inf->apply_config;

{
  my $rcfile = path('./perlcritic.rc')->openw_utf8;
  $rcfile->print( $config->as_ini, "\n" );
  close $rcfile or croak 'Something fubared closing perlcritic.rc';
}
my $deps = $inf->own_deps;
{
  my $target = path('./misc');
  $target->mkpath if not $target->is_dir;

  my $depsfile = $target->child('perlcritic.deps')->openw_utf8;
  for my $key ( sort keys %{$deps} ) {
    $depsfile->printf( "%s~%s\n", $key, $deps->{$key} );
    *STDERR->printf( "%s => %s\n", $key, $deps->{$key} );
  }
  close $depsfile or carp 'Something fubared closing perlcritic.deps';
}



Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.009.
use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Add a baked version of your configuration to tree automatically",
  "AUTHOR" => "Kent Fredric <kentnl\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Dist-Zilla-Plugin-INI-Baked",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "Dist::Zilla::Plugin::INI::Baked",
  "PREREQ_PM" => {
    "Dist::Zilla::File::FromCode" => 0,
    "Dist::Zilla::Role::FileGatherer" => 0,
    "Dist::Zilla::Util::CurrentCmd" => "0.002000",
    "Dist::Zilla::Util::ExpandINI" => "0.001001",
    "MooX::Lsub" => 0,
    "Moose" => 0,
    "Path::Tiny" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Test::DZil" => 0,
    "Test::More" => 0,
    "Test::Requires" => 0
  },
  "VERSION" => "0.002003",
  "test" => {
    "TESTS" => "t/*.t t/00-compile/*.t"
  }
);


my %FallbackPrereqs = (
  "Dist::Zilla::File::FromCode" => 0,
  "Dist::Zilla::Role::FileGatherer" => 0,
  "Dist::Zilla::Util::CurrentCmd" => "0.002000",
  "Dist::Zilla::Util::ExpandINI" => "0.001001",
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "MooX::Lsub" => 0,
  "Moose" => 0,
  "Path::Tiny" => 0,
  "Test::DZil" => 0,
  "Test::More" => 0,
  "Test::Requires" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
